internal/connector/pingone_platform/pingone_platform_connector.go

```diff
@@ -0,0 +1,111 @@
+package pingone_platform
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"text/template"
+
+	sdk "github.com/patrickcping/pingone-go-sdk-v2/pingone"
+	"github.com/pingidentity/pingctl/internal/connector"
+	"github.com/pingidentity/pingctl/internal/connector/pingone_platform/resources"
+	"github.com/pingidentity/pingctl/internal/logger"
+)
+
+const (
+	ServiceName = "pingone-platform"
+)
+
+// Verify that the connector satisfies the expected interfaces
+var (
+	_ connector.Exportable      = &PingonePlatformConnector{}
+	_ connector.Authenticatable = &PingonePlatformConnector{}
+)
+
+type PingonePlatformConnector struct {
+	context       context.Context
+	apiClient     *sdk.Client
+	environmentID string
+}
+
+// Utility method for creating a PingonePlatformConnector
+func Connector(ctx context.Context, apiClient *sdk.Client, environmentID string) *PingonePlatformConnector {
+	return &PingonePlatformConnector{
+		context:       ctx,
+		apiClient:     apiClient,
+		environmentID: environmentID,
+	}
+}
+
+func (c *PingonePlatformConnector) Export(format, outputDir string, overwriteExport bool) error {
+	l := logger.Get()
+
+	l.Debug().Msgf("Exporting all PingOne Platform Resources...")
+
+	hclImportBlockTemplate, err := template.New("HCLImportBlock").Parse(
```

Comment:
If we end up with a lot of templates in the future (which we may not for this tool) it should be possible to move them to separate files and embed them in the binary that gets built.

Reply:
Embedded this template in exportable.go 👍